api/lifetime_watcher.go

```diff
@@ -290,42 +292,67 @@ func (r *LifetimeWatcher) doRenew() error {
 					}
 				}
 
-				leaseDuration = fallbackLeaseDuration
+				// Calculate remaining duration until initial token lease expires
+				leaseDuration = initialTime.Add(time.Duration(initLeaseDuration) * time.Second).Sub(time.Now())
+				if errorBackoff == nil {
+					errorBackoff = &backoff.ExponentialBackOff{
+						MaxElapsedTime:      leaseDuration,
+						RandomizationFactor: backoff.DefaultRandomizationFactor,
+						InitialInterval:     initialRetryInterval,
+						MaxInterval:         5 * time.Minute,
+						Multiplier:          2,
+						Clock:               backoff.SystemClock,
+					}
+					errorBackoff.Reset()
+				}
 				break
 			}
+			errorBackoff = nil
 
 			// Push a message that a renewal took place.
 			select {
 			case r.renewCh <- &RenewOutput{time.Now().UTC(), renewal}:
 			default:
 			}
 
+			// Reset initial time
+			initialTime = time.Now()
```

Comment:
It might be nice to combine this with the block of code resetting the initial lease duration below (L328), as they are logically very closely related.

Reply:
Done.